File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void flatten(TreeNode root) {
        if (root == null) {
            return;
        }

        // 왼쪽 서브트리를 오른쪽으로 이동
        if (root.left != null) {
            flatten(root.left);

            TreeNode temp = root.right;
            root.right = root.left;
            root.left = null;

            // 오른쪽으로 이동한 노드의 마지막에 오른쪽 서브트리를 연결
            while (root.right != null) {
                root = root.right;
            }
            root.right = temp;
        }

        // 오른쪽 서브트리를 펼치기
        flatten(root.right);
    }
}

File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def flatten(self, root):
        """
        :type root: TreeNode
        :rtype: None Do not return anything, modify root in-place instead.
        """
        if not root:
            return None

        # 왼쪽 서브트리를 오른쪽으로 이동
        if root.left:
            self.flatten(root.left)

            temp = root.right
            root.right = root.left
            root.left = None

            # 오른쪽으로 이동한 노드의 마지막에 오른쪽 서브트리를 연결
            while root.right:
                root = root.right
            root.right = temp

        # 오른쪽 서브트리를 펼치기
        self.flatten(root.right)